tests/unit/test__pandas_helpers.py

```diff
@@ -1093,33 +1077,6 @@ def test_dataframe_to_parquet_compression_method(module_under_test):
     assert call_args.kwargs.get("compression") == "ZSTD"
 
 
-@pytest.mark.skipif(pandas is None, reason="Requires `pandas`")
-def test_dataframe_to_bq_schema_fallback_needed_wo_pyarrow(module_under_test):
-    dataframe = pandas.DataFrame(
-        data=[
-            {"id": 10, "status": u"FOO", "execution_date": datetime.date(2019, 5, 10)},
-            {"id": 20, "status": u"BAR", "created_at": datetime.date(2018, 9, 12)},
-        ]
-    )
-
-    no_pyarrow_patch = mock.patch(module_under_test.__name__ + ".pyarrow", None)
-
-    with no_pyarrow_patch, warnings.catch_warnings(record=True) as warned:
-        detected_schema = module_under_test.dataframe_to_bq_schema(
-            dataframe, bq_schema=[]
-        )
-
-    assert detected_schema is None
-
-    # a warning should also be issued
-    expected_warnings = [
-        warning for warning in warned if "could not determine" in str(warning).lower()
```

Comment:
Looks like the warning for this isn't issued anymore. I wonder if it's possible to get a dataframe without detectable schema if pyarrow is installed?

Reply:
This particular test covered the [case](https://github.com/googleapis/python-bigquery/blob/e58702967d572e83b4c774278818302594a511b7/google/cloud/bigquery/_pandas_helpers.py#L346-L352) when we couldn't even start detecting the schema with `pyarrow` because the later is missing. This is not possible anymore, as `pyarrow` is now required, thus that particular warning has been removed.

However, we still call [augment_schema()](https://github.com/googleapis/python-bigquery/blob/e58702967d572e83b4c774278818302594a511b7/google/cloud/bigquery/_pandas_helpers.py#L354-L411) immediately after the deleted lines. That function can also issue a warning on its own if schema detection fails for some fields, but those paths are covered by other tests.